File: backend/src/main/scala/kpn/api/common/LocationChangesInfo.scala
package kpn.api.common

case class LocationChangesInfo(
  routeType: RouteType,
  locationInfos: Seq[LocationInfo],
  routeChanges: ChangeSetElementRefs,
  nodeChanges: ChangeSetElementRefs,
  happy: Boolean,
  investigate: Boolean
)

File: backend/api/src/main/java/kpn/api/common/LocationChangesInfo.java
package kpn.api.common;

import kpn.api.common.ChangeSetElementRefs;
import kpn.api.common.LocationInfo;
import kpn.api.common.RouteType;

import com.google.common.collect.ImmutableList;

public record LocationChangesInfo(
  RouteType routeType,
  ImmutableList<LocationInfo> locationInfos,
  ChangeSetElementRefs routeChanges,
  ChangeSetElementRefs nodeChanges,
  Boolean happy,
  Boolean investigate
) {
}
